src/util.rs

```diff
@@ -16,6 +18,17 @@ use crate::{
     KEY_WIDTH,
 };
 
+/// Check if a process if using port 80. A return value of false means there wasn't a process
+/// found.
+pub fn check_for_existing_server() -> Result<bool> {
+    // If it was unsuccessful (i.e. exit code 1), then nothing is currently using that port.
+    Ok(Command::new("lsof")
```

Comment:
Mh... is there maybe a better way to check this than `lsof`? From what I've read it doesn't come shipped with every distribution. What do you think about using `ps aux` and `grep`?

Reply:
Yeah the portability part of this is tough. Unfortunately I don't think `ps` will list any open ports that a process is using. Unless I misread any options on the man page!

We can make it more robust though and try using another tool if `lsof` fails. `ss` is another good option but not every distribution will have it I'm sure.

Reply:
Think a better way is to just try and initiate a TCP connection using the standard library. I pushed that in the subsequent CR and can squash my commits if you'd like.

Reply:
Nice! That sounds like a good solution!